runtime/tchannel_server.go

```diff
@@ -98,34 +98,42 @@ func NewTChannelRouter(registrar tchannel.Registrar, g *Gateway) *TChannelRouter
 func (s *TChannelRouter) Register(
 	endpointID, handlerID, method string,
 	h TChannelHandler,
+	fields []zapcore.Field,
```

Comment:
This method is getting completely out of hand, the `register` semantic is simple, and yet now it is bloated with 6 arguments and returns an endpoint? It does way more than what its name means.

Reply:
Taking an options struct is fine. 

You could also pass in the `Endpoint` similar to the `Router` API for http.

Reply:
Passing a `TChannelEndpoint` to `Register` like `Router` for http.